dissect/database/sqlite3/sqlite3.py

```diff
@@ -49,17 +50,59 @@
 
 SQLITE3_HEADER_MAGIC = b"SQLite format 3\x00"
 
-WAL_HEADER_MAGIC_LE = 0x377F0682
-WAL_HEADER_MAGIC_BE = 0x377F0683
-WAL_HEADER_MAGIC = {WAL_HEADER_MAGIC_LE, WAL_HEADER_MAGIC_BE}
-
 
 class SQLite3:
-    def __init__(self, fh: BinaryIO, wal_fh: BinaryIO | None = None):
+    def __init__(
+        self,
+        fh: Path | BinaryIO,
+        wal_fh: Path | BinaryIO | None = None,
+        wal_checkpoint: Checkpoint | int | None = None,
```

Comment:
```suggestion
        wal: WAL | Path | BinaryIO | None = None,
        checkpoint: Checkpoint | int | None = None,
```
Maybe a bit cleaner arguments.